lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py

```diff
@@ -270,6 +271,7 @@ def update_grid_finder(self, aux_trans=None, **kwargs):
         self.grid_finder.update(**kwargs)
         self._old_limits = None  # Force revalidation.
 
+    @_api.delete_parameter("3.9", "nth_coord")
```

Comment:
Should we not pass _nth_coord_ticks_ to `FixedAxisArtistHelper` below?  Then it will just be set to what was inferred by the parent class
https://github.com/matplotlib/matplotlib/blob/4846a8b3a4461c3761d414de67d980444a20e12e/lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py#L59-L60

I think it we allow it to be passed down then we can still end up with something that is inconsistent with _loc_.

Reply:
Ah, good catch, the deprecation in the curvilinear case was too much, as for curvilinear axes (not floating axes) it does actually make sense to say, draw the first coordinate ticks on the vertical (second) axis, see e.g. the second example of https://matplotlib.org/stable/gallery/axisartist/demo_curvelinear_grid.html.  Adjusted everything accordingly.